src/extensions/default/Inline3DParametersEditor/ParameterEditor.js

```diff
@@ -0,0 +1,149 @@
+        this.$element = $(Mustache.render(Template, Strings));
+        $parent.append(this.$element);
+
+        this._callback = callback;
+
+        this._handleXValueSliderDrag = this._handleXValueSliderDrag.bind(this);
+        this._handleYValueSliderDrag = this._handleYValueSliderDrag.bind(this);
+        this._handleZValueSliderDrag = this._handleZValueSliderDrag.bind(this);
+        this.$xValueSlider = this.$element.find("#_x");
+        this.$yValueSlider = this.$element.find("#_y");
+        this.$zValueSlider = this.$element.find("#_z");
+        this._x = parseFloat(this.$xValueSlider.val());
+        this._y = parseFloat(this.$yValueSlider.val());
+        this._z = parseFloat(this.$zValueSlider.val());
+
+        this.$selectionBase = this.$element.find(".selector-base");
+
+        // Attach event listeners to main UI elements
+        this._addListeners();
+    }
+
+
+    /** Returns the root DOM node of the UI */
+    ParameterEditor.prototype.getRootElement = function () {
+        return this.$element;
+    };
+
+    ParameterEditor.prototype._addListeners = function () {
+        var self = this;
+        this.$xValueSlider.mousedown(function(e) {
+            self._x = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$yValueSlider.mousedown(function(e) {
+            self._y = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$zValueSlider.mousedown(function(e) {
+            self._z = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        
+        this.$xValueSlider.change(function(e) {
+            self._commitParameters(self.$xValueSlider.val() + " 3 0");
+        });
+
+        this._registerDragHandler(this.$xValueSlider, this._handleXValueSliderDrag);
+        this._registerDragHandler(this.$yValueSlider, this._handleYValueSliderDrag);
+        this._registerDragHandler(this.$zValueSlider, this._handleZValueSliderDrag);
+
+    };
+
+    ParameterEditor.prototype.focus = function () {
+        if (!this.$selectionBase.is(":focus")) {
+            this.$selectionBase.focus();
+            return true;
+        }
+        return false;
+    };
+
+    ParameterEditor.prototype._commitParameters = function (parameters) {
+        this._callback(parameters);
+    };
+
+    ParameterEditor.prototype.setParametersFromString = function (parameters) {
+        this.setSliderProperties(parameters);
+        this._commitParameters(parameters, true);
+    };
+
+    ParameterEditor.prototype.isValidSetOfParameters = function (parameters) {
+        var ParameterRegex = new RegExp(/((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?)/g);
+        return ParameterRegex.test(parameters);
+    };
+
+    ParameterEditor.prototype.setSliderProperties = function (parameters) {
+        var parametersArray = parameters.split(" ");
```

Comment:
Made it split using regex, also it can have spaces before the first and after the last parameter. Fixed that too.
[Commit](https://github.com/mozilla/brackets/pull/651/commits/d7bc0d7a3203a73f6d57df4b45451e1d437b8bf4)